### ISSUE TITLE
Update OAI endpoint and fields

### ISSUE THREAD

**Issue by kjallen** [scientist-softserv/oral-history#42]
# Story

The location of the OAI provider and some fields have changed slightly. As a result, some code will either need to be updated or moved to configuration files, including the root url of the [OAI provider.](https://github.com/scientist-softserv/oral-history/blob/e45a4760b7f9f0dc3944602280ebdeaa5510779c/app/models/oral_history_item.rb#L29)

Examples of the new url:

Single record:
https://oh-staff.library.ucla.edu/oai/?verb=GetRecord&identifier=21198/zz002kpmhj

All records:
https://oh-staff.library.ucla.edu/oai/?verb=ListRecords

Root url:
`https://oh-staff.library.ucla.edu/oai/`

Verbs supported:
`GetRecord` with `identifier`
`ListRecords` (no parameters required)

No longer required:
`metadata_prefix`
`set` (default is now oral history)
No redemption token is needed, all results should be returned with ListRecords (request takes about 60-70 seconds for generation, factor into time out on Ruby side)

The `identifier` also has been switched to use the database stored format, with a slash instead of a dash (ie `21198/zz002kpmhj` vs `21198-zz002kpmhj` (old format, should no longer be used)

# Acceptance Criteria

- [oral_history_item.rb](https://github.com/scientist-softserv/oral-history/blob/e45a4760b7f9f0dc3944602280ebdeaa5510779c/app/models/oral_history_item.rb#L29) and other relevant files updated to reflect new OAI location
- Harvest and index tested to confirm fields are correctly picked up
- Wav files are generated

# Testing Instructions and Sample Files

Single record:
https://oh-staff.library.ucla.edu/oai/?verb=GetRecord&identifier=21198/zz002kpmhj

All records:
https://oh-staff.library.ucla.edu/oai/?verb=ListRecords

**Comment by labradford** [comment]
- [ ] Update OAI url
- [ ] Remove extra headers
- [ ] build in a wait function to avoid timeout `Faraday::TimeoutError (Net::ReadTimeout with #<TCPSocket:(closed)>)`
- [ ] update def self.fetch_first_id method
- [ ] replace `/` in id with `-` (URL for item page will not support `/`)
- [ ] verify all metadata fields are added
- [ ] verify waveforms are generated

**Comment by kjallen** [comment]
hi @labradford, wanted to check in -- looking at the tasks above, I don't *think* folks are waiting on anything from the UCLA side, but wanted to confirm we weren't the blocker. Is there any additional info you need from me to confirm or clarify?

**Comment by labradford** [comment]
> hi @labradford, wanted to check in -- looking at the tasks above, I don't _think_ folks are waiting on anything from the UCLA side, but wanted to confirm we weren't the blocker. Is there any additional info you need from me to confirm or clarify?

This was just a list of things that we would need to do to implement the new feed. I don't believe there is anything we need from UCLA at this time

**Comment by aprilrieger** [comment]
We pulled out the requirement:
- [ ] full text import is being indexed properly (is not working with new feed, will need to troubleshoot)
As this will be moved to the i17 work once the updated feed gets merged into main.
https://github.com/scientist-softserv/oral-history/issues/17
